src/modules/ethereum/transactions/gasPrices/optimistic/fee.ts

```diff
@@ -74,12 +65,19 @@ export async function createOptimisticFee({
         eip1559GasPrice.priorityFee,
         eip1559GasPrice.maxFee,
         gas,
-        to,
-        value,
-        data,
+        to ?? '0x',
+        valueToHex(value ?? '0x0'),
+        data ?? '0x',
       ])
     : classicGasPrice != null
-    ? rlpEncode([nonce, classicGasPrice, gas, to, value, data])
+    ? rlpEncode([
+        nonce,
+        classicGasPrice,
+        gas,
+        to ?? '0x',
```

Comment:
May be if is better to define to, value and data outside to avoid these 2 same sets of lines?

Reply:
you're right